### PR TITLE
Fix RWKV backward on GPU

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

Fixes the backward pass for RWKV on GPU. The function backward was not adapted to the revamp of the forward, my bad.

### CODE REVIEW

File: src/transformers/models/rwkv/modeling_rwkv.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def forward(ctx, time_decay, time_first, key, value, state=None, return_state=Fa
 
     @staticmethod
     # g stands for grad
-    def backward(ctx, g_output):
+    def backward(ctx, g_output, g_state=None):
```
Comment (younesbelkada) [inline src/transformers/models/rwkv/modeling_rwkv.py:162]: why the variable is not used later on?

File: src/transformers/models/rwkv/modeling_rwkv.py [hunk 2]
```diff
@@ -188,17 +188,14 @@ def backward(ctx, g_output):
             g_key,
             g_value,
         )
-        g_time_decay = torch.sum(g_time_decay, dim=0)
-        g_time_first = torch.sum(g_time_first, dim=0)
 
         return (
-            None,
-            None,
-            None,
```
Comment (younesbelkada) [inline src/transformers/models/rwkv/modeling_rwkv.py:197]: also just out of curiosity why the number of outputs changed ? 🤔

**Review by younesbelkada — APPROVED** [review]
Thank you very much for fixing! I just left 2 questions for learning purpose!

File: src/transformers/models/rwkv/modeling_rwkv.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/models/rwkv/modeling_rwkv.py:162]: Because we don't handle the gradient of the state. But autograd is not happy if this isn't here.

File: src/transformers/models/rwkv/modeling_rwkv.py (hunk 2 above)
Comment (sgugger) [reply src/transformers/models/rwkv/modeling_rwkv.py:197]: Because autograd wants one gradient per input of the forward :-)
It used to have three ints then the variables, but no state and no bool, this PR adapts it to the changes I made.

**Review by ArthurZucker — APPROVED** [review]
Wow thanks for fixing, and showing us how it's done! 🔥

File: src/transformers/models/rwkv/modeling_rwkv.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/models/rwkv/modeling_rwkv.py:162]: Thanks so much for explaning!
